Segmentation fault at incoming connection

**Original report ([archived issue](https://osrf-migration.github.io/gzweb-gh-pages/#!/osrf/gzweb/issues/155)) by Anonymous.**

----------------------------------------
I am trying to use gazebo server and gzweb.
I have installed gazebo on ubuntu via gazebo.sh
After that I have installed gzweb as described here: http://gazebosim.org/gzweb.html#install-collapse-2

Steps to reproduce:

1. gzserver

2. npm start

3. Open localhost:8080 in browser

After that segmentation fault occures

Log of npm start below:
```
pam@DESKTOP-T6FUHBJ:/mnt/d/work/gz-ws/gzweb$ npm start

> gzweb@1.3.0 start /mnt/d/work/gz-ws/gzweb
> if [ $npm_config_port ]; then port=$npm_config_port; fi; cd gzbridge && ./server.js $port

Tue Jul 03 2018 15:46:37 GMT+0300 (STD) Static server listening on port: 8080
--------------------------------------------------------------
Gazebo transport node connected to gzserver.
Pose message filter parameters between successive messages:
  minimum seconds: 0.02
  minimum XYZ distance squared: 0.00001
  minimum Quartenion distance squared: 0.00001
--------------------------------------------------------------
Tue Jul 03 2018 15:46:43 GMT+0300 (STD) New connection accepted from: http://localhost:8080 ::1
Tue Jul 03 2018 15:46:44 GMT+0300 (STD) Received Message: {"op":"subscribe","id":"subscribe:~/status:1","type":"status","topic":"~/status","compression":"none","throttle_rate":0} from http://localhost:8080 ::1
Tue Jul 03 2018 15:46:44 GMT+0300 (STD) Received Message: {"op":"subscribe","id":"subscribe:~/material:2","type":"material","topic":"~/material","compression":"none","throttle_rate":0} from http://localhost:8080 ::1
Tue Jul 03 2018 15:46:44 GMT+0300 (STD) Received Message: {"op":"subscribe","id":"subscribe:~/scene:3","type":"scene","topic":"~/scene","compression":"none","throttle_rate":0} from http://localhost:8080 ::1
Tue Jul 03 2018 15:46:44 GMT+0300 (STD) Received Message: {"op":"subscribe","id":"subscribe:~/physics:4","type":"physics","topic":"~/physics","compression":"none","throttle_rate":0} from http://localhost:8080 ::1
Tue Jul 03 2018 15:46:44 GMT+0300 (STD) Received Message: {"op":"subscribe","id":"subscribe:~/pose/info:5","type":"pose","topic":"~/pose/info","compression":"none","throttle_rate":0} from http://localhost:8080 ::1
Tue Jul 03 2018 15:46:44 GMT+0300 (STD) Received Message: {"op":"subscribe","id":"subscribe:~/request:6","type":"request","topic":"~/request","compression":"none","throttle_rate":0} from http://localhost:8080 ::1
Tue Jul 03 2018 15:46:44 GMT+0300 (STD) Received Message: {"op":"subscribe","id":"subscribe:~/model/info:7","type":"model","topic":"~/model/info","compression":"none","throttle_rate":0} from http://localhost:8080 ::1
Tue Jul 03 2018 15:46:44 GMT+0300 (STD) Received Message: {"op":"subscribe","id":"subscribe:~/visual:8","type":"visual","topic":"~/visual","compression":"none","throttle_rate":0} from http://localhost:8080 ::1
Tue Jul 03 2018 15:46:44 GMT+0300 (STD) Received Message: {"op":"subscribe","id":"subscribe:~/world_stats:9","type":"world_stats","topic":"~/world_stats","compression":"none","throttle_rate":0} from http://localhost:8080 ::1
Segmentation fault (core dumped)

npm ERR! Linux 4.4.0-17134-Microsoft
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "start"
npm ERR! node v4.2.6
npm ERR! npm  v3.5.2
npm ERR! code ELIFECYCLE
npm ERR! gzweb@1.3.0 start: `if [ $npm_config_port ]; then port=$npm_config_port; fi; cd gzbridge && ./server.js $port`
npm ERR! Exit status 139
npm ERR!
npm ERR! Failed at the gzweb@1.3.0 start script 'if [ $npm_config_port ]; then port=$npm_config_port; fi; cd gzbridge && ./server.js $port'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the gzweb package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     if [ $npm_config_port ]; then port=$npm_config_port; fi; cd gzbridge && ./server.js $port
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs gzweb
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls gzweb
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /mnt/d/work/gz-ws/gzweb/npm-debug.log
```

--- a/gzbridge/GazeboInterface.cc
+++ b/gzbridge/GazeboInterface.cc
@@ -710,7 +710,7 @@ void GazeboInterface::ProcessMessages()
           // simulate latching stats topic
           if (this->statsMsgs.empty() && this->statsMsg.get() != nullptr)
           {
-              this->statsMsgs.push_back(this->statsMsg);
+            this->statsMsgs.push_back(this->statsMsg);
           }
         }
       }

